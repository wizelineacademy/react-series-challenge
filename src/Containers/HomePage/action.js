export const QUERY_TRENDING = 'QUERY_TRENDING';
export const QUERY_TRENDING_SUCCESS = 'QUERY_TRENDING_SUCCESS';
export const QUERY_TRENDING_ERROR = 'QUERY_TRENDING_ERROR';

export function queryTrending(param) {
  return {
    type: QUERY_TRENDING,
    param
  };
}

export function queryTrendingSuccess(data) {
  return {
    type: QUERY_TRENDING_SUCCESS,
    data
  };
}

export function queryTrendingError(error) {
  return {
    type: QUERY_TRENDING_ERROR,
    error
  };
}