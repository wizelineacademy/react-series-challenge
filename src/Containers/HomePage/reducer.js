import {
  QUERY_TRENDING,
  QUERY_TRENDING_ERROR,
  QUERY_TRENDING_SUCCESS
} from './action';

const initialState = {
  trendingGifs: [],
  trendingLoading: true
};

function trendingReducer(state = initialState, action) {
  
  switch (action.type) {
    case QUERY_TRENDING:
      return {...state,trendingLoading:true}
    case QUERY_TRENDING_SUCCESS:
      const { data } = action.data;
        return {...state,trendingLoading:false,trendingGifs:data};
    case QUERY_TRENDING_ERROR:
      return {...state,trendingGifs:null, trendingLoading:false}
    default:
      return state;
  }
}

export default trendingReducer;