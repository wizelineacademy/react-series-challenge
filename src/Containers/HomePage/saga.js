import {
  queryTrendingError,
  queryTrendingSuccess,
  QUERY_TRENDING
} from './action'
import serialize from '../../utils/serialize'
import { call, put, takeLatest } from 'redux-saga/effects'
import axios from 'axios'

function* doQueryTrending({ param = {} }) {

  const query = serialize({
    limit: 10,
    api_key: 'Y354pcUIVTGZXofCIqvlYWA2Bv1khCCZ'
  });
  const response = yield call([axios, 'get'], `https://api.giphy.com/v1/gifs/trending?${query}`);
  if (response.error) {
    yield put(queryTrendingError(response.error));
    return;
  }
  yield put(queryTrendingSuccess(response.data));
}


export default function* queryTrendingWatcher() {
  yield takeLatest(QUERY_TRENDING, doQueryTrending);
}