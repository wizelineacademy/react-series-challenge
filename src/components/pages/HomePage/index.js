import React from "react";
import PublicLayout from "../../layouts/PublicLayout/index";
import TrendingContainer from "../../containers/TrendingContainer/index";

const HomePage = () => {
  return (
    <PublicLayout>
      <h3>Click to Add a Favorite</h3>
      <TrendingContainer />
    </PublicLayout>
  );
};

export default HomePage;
