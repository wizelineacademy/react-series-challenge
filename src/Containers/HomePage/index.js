import React, { Component } from 'react'
import { queryTrending } from './action'
import { connect } from 'react-redux'
import Gif from "../../components/Gifs"

class HomePage extends Component {
  state = {
    openFullScreen: false,
    selectedImageUrl: ''
  }

  componentDidMount() {
    const { queryTrending } = this.props;
    queryTrending();
  }

  render() {
    const { gifs} = this.props;
    return (
        <React.Fragment>
        {
          gifs && gifs.map((gif, index) => {return <Gif url={gif.images.fixed_width.url} id={gif.id} key={gif.id}/>})
        }
        </React.Fragment>
    )
  }
}

const mapStateToProps = state => {
  return{
  gifs: state.home.trendingGifs,
  loading: state.home.trendingLoading
}}

const mapDispatchToProps = {
  queryTrending
}

export default connect(mapStateToProps, mapDispatchToProps)(HomePage);