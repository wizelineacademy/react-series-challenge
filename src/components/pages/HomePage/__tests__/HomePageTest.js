import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import HomePage from "../index";

describe("HomePage", () => {
  it("should render without crashing", () => {
    const testHomePage = shallow(<HomePage />);

    expect(shallowToJson(testHomePage)).toMatchSnapshot();
  });
});
