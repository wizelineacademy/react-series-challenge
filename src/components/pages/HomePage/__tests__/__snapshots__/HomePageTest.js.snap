// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render without crashing 1`] = `
<PublicLayout>
  <h3>
    Click to Add a Favorite
  </h3>
  <Connect(TrendingContainer) />
</PublicLayout>
`;
